src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexRunnerFactory.cs

```diff
@@ -22,6 +22,20 @@ public SymbolicRegexRunnerFactory(RegexTree regexTree, RegexOptions options, Tim
             var converter = new RegexNodeConverter(bddBuilder, regexTree.CaptureNumberSparseMapping);
 
             SymbolicRegexNode<BDD> rootNode = converter.ConvertToSymbolicRegexNode(regexTree.Root);
+            // Determine if the root node is supported for safe handling
+            int threshold = SymbolicRegexThresholds.GetSymbolicRegexSafeSizeThreshold();
+            Debug.Assert(threshold > 0);
+
+            // Skip the threshold check if the threshold equals int.MaxValue
```

Comment:
NIT: in other parts of the runtime, we usually signal `-1` to be the "infinite" value. In Regex that is what we use to describe infinite timeout as well. should we use `-1` as the value to skip the check?

Reply:
Since the value `int.MaxValue` is also used as "infinity" for the upper bound and the lower and upper bound values are in general assumed to be >=0 in all debug asserts, so giving -1 a special status here is questionable to me.

Reply:
But at the same time, it is not actually directly tied to the interpretation of the lower and upper bounds in loops, so I'm fine either way.

Reply:
The change that would allow `-1` as special value for no bound could be the following in `SymbolicRegexThresholds`:
If I get agreement from @stephentoub , I would go ahead and apply this change
```
        internal static int GetSymbolicRegexSafeSizeThreshold()
        {
            object? safeSizeThreshold = AppContext.GetData(SymbolicRegexSafeSizeThreshold_ConfigKeyName);

            return (safeSizeThreshold is not int safeSizeThresholdInt || safeSizeThresholdInt == 0 || safeSizeThresholdInt < -1) ?
                DefaultSymbolicRegexSafeSizeThreshold :
                (safeSizeThresholdInt == -1 ? int.MaxValue : safeSizeThresholdInt);
        }
```